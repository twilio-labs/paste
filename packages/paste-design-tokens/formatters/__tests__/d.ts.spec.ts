import * as theo from 'theo';
import {resolve} from 'path';
import {dTSTokenFormat} from '../d.ts';

theo.registerFormat('d.ts', dTSTokenFormat);

describe('dTSFormatter', () => {
  it('should return typescript definitions formatted tokens', () => {
    return theo
      .convert({
        transform: {
          type: 'web',
          file: resolve(__dirname, '../__fixtures__/index.yml'),
        },
        format: {
          // @ts-ignore Theo isn't typed for custom format types
          type: 'd.ts',
        },
      })
      .then((dTS: string) => {
        expect(dTS).toMatchSnapshot();
        return true;
      })
      .catch((error: string) => console.log(`Something went wrong: ${error}`));
  });
});
