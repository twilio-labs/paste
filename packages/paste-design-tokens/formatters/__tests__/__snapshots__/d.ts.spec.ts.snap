// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dTSFormatter should return typescript definitions formatted tokens 1`] = `
"export declare const colorBackgroundBody = \\"rgb(255, 255, 255)\\";
export declare const colorBackgroundBrandHighlight = \\"rgb(242, 47, 70)\\";
export declare const colorBackgroundBrandHighlightDark = \\"rgb(235, 15, 41)\\";
export declare const colorBackgroundBrandHighlightDarker = \\"rgb(194, 37, 55)\\";
export declare const colorBackgroundBrandPrimary = \\"rgb(13, 18, 43)\\";
export declare const colorBackgroundDefault = \\"rgb(246, 246, 246)\\";
export declare const colorBackgroundElementSelection = \\"rgb(236, 187, 193)\\";
export declare const colorBackgroundError = \\"rgb(206, 80, 95)\\";
export declare const colorBackgroundErrorDark = \\"rgb(174, 33, 49)\\";
export declare const colorBackgroundErrorLight = \\"rgb(194, 37, 55)\\";
export declare const colorBackgroundRowActive = \\"rgb(230, 230, 230)\\";
export declare const colorBackgroundRowFocus = \\"rgb(214, 214, 214)\\";
export declare const colorBackgroundRowHover = \\"rgb(214, 214, 214)\\";
export declare const colorBackgroundRowSelected = \\"rgb(230, 230, 230)\\";
export declare const colorBackgroundRowStriped = \\"rgb(235, 235, 235)\\";
export declare const colorBackgroundSuccess = \\"rgb(100, 209, 138)\\";
export declare const colorBackgroundSuccessDark = \\"rgb(80, 167, 110)\\";
export declare const colorBackgroundSuccessLight = \\"rgb(162, 227, 184)\\";
export declare const colorBackgroundWarning = \\"rgb(251, 208, 144)\\";
export declare const colorBackgroundWarningDark = \\"rgb(199, 141, 56)\\";
export declare const colorBackgroundWarningLight = \\"rgb(249, 177, 71)\\";
export declare const colorBrandHighlight = \\"rgb(242, 47, 70)\\";
export declare const colorBrandHighlightDark = \\"rgb(235, 15, 41)\\";
export declare const colorBrandPrimary = \\"rgb(13, 18, 43)\\";
export declare const colorGray1 = \\"rgb(255, 255, 255)\\";
export declare const colorGray10 = \\"rgb(34, 34, 34)\\";
export declare const colorGray2 = \\"rgb(246, 246, 246)\\";
export declare const colorGray3 = \\"rgb(235, 235, 235)\\";
export declare const colorGray4 = \\"rgb(230, 230, 230)\\";
export declare const colorGray5 = \\"rgb(214, 214, 214)\\";
export declare const colorGray6 = \\"rgb(204, 204, 204)\\";
export declare const colorGray7 = \\"rgb(153, 153, 153)\\";
export declare const colorGray8 = \\"rgb(102, 102, 102)\\";
export declare const colorGray9 = \\"rgb(68, 68, 68)\\";
export declare const colorTextBrandHighlight = \\"rgb(242, 47, 70)\\";
export declare const colorTextBrandHighlightDark = \\"rgb(235, 15, 41)\\";
export declare const colorTextBrandPrimary = \\"rgb(255, 255, 255)\\";
export declare const colorTextDefault = \\"rgb(68, 68, 68)\\";
export declare const colorTextError = \\"rgb(194, 37, 55)\\";
export declare const colorTextIconDefault = \\"rgb(68, 68, 68)\\";
export declare const colorTextInverse = \\"rgb(255, 255, 255)\\";
export declare const colorTextInverseWeak = \\"rgb(235, 235, 235)\\";
export declare const colorTextLabel = \\"rgb(68, 68, 68)\\";
export declare const colorTextLink = \\"rgb(0, 112, 204)\\";
export declare const colorTextLinkActive = \\"rgb(0, 89, 163)\\";
export declare const colorTextLinkDestructive = \\"rgb(194, 37, 55)\\";
export declare const colorTextLinkDestructiveActive = \\"rgb(155, 29, 44)\\";
export declare const colorTextLinkDestructiveDisabled = \\"rgb(236, 187, 193)\\";
export declare const colorTextLinkDestructiveFocus = \\"rgb(174, 33, 49)\\";
export declare const colorTextLinkDestructiveHover = \\"rgb(174, 33, 49)\\";
export declare const colorTextLinkDisabled = \\"rgb(102, 169, 224)\\";
export declare const colorTextLinkFocus = \\"rgb(0, 97, 192)\\";
export declare const colorTextLinkHover = \\"rgb(0, 97, 192)\\";
export declare const colorTextPlaceholder = \\"rgb(102, 102, 102)\\";
export declare const colorTextSuccess = \\"rgb(100, 209, 138)\\";
export declare const colorTextWarning = \\"rgb(249, 177, 71)\\";
export declare const colorTextWeak = \\"rgb(230, 230, 230)\\";
export declare const fontSize0 = \\"0.6875rem\\";
export declare const fontSize1 = \\"0.75rem\\";
export declare const fontSize2 = \\"0.8125rem\\";
export declare const fontSize3 = \\"0.875rem\\";
export declare const fontSize4 = \\"0.9375rem\\";
export declare const fontSize5 = \\"1rem\\";
export declare const fontSize6 = \\"1.125rem\\";
export declare const fontSize7 = \\"1.25rem\\";
export declare const fontSize8 = \\"1.875rem\\";
export declare const fontSize9 = \\"2.25rem\\";
export declare const space0 = \\"0.25rem\\";
export declare const space1 = \\"0.5rem\\";
export declare const space10 = \\"2.75rem\\";
export declare const space11 = \\"3rem\\";
export declare const space12 = \\"3.25rem\\";
export declare const space13 = \\"3.5rem\\";
export declare const space14 = \\"3.75rem\\";
export declare const space15 = \\"4rem\\";
export declare const space2 = \\"0.75rem\\";
export declare const space3 = \\"1rem\\";
export declare const space4 = \\"1.25rem\\";
export declare const space5 = \\"1.5rem\\";
export declare const space6 = \\"1.75rem\\";
export declare const space7 = \\"2rem\\";
export declare const space8 = \\"2.25rem\\";
export declare const space9 = \\"2.5rem\\";
export declare const zIndex0 = \\"0\\";
export declare const zIndex10 = \\"10\\";
export declare const zIndex20 = \\"20\\";
export declare const zIndex30 = \\"30\\";
export declare const zIndex40 = \\"40\\";
export declare const zIndex50 = \\"50\\";
export declare const zIndex60 = \\"60\\";
export declare const zIndex70 = \\"70\\";
export declare const zIndex80 = \\"80\\";
export declare const zIndex90 = \\"90\\";
export declare const backgroundColors: {
  colorBackgroundBody: \\"rgb(255, 255, 255)\\";
  colorBackgroundBrandHighlight: \\"rgb(242, 47, 70)\\";
  colorBackgroundBrandHighlightDark: \\"rgb(235, 15, 41)\\";
  colorBackgroundBrandHighlightDarker: \\"rgb(194, 37, 55)\\";
  colorBackgroundBrandPrimary: \\"rgb(13, 18, 43)\\";
  colorBackgroundDefault: \\"rgb(246, 246, 246)\\";
  colorBackgroundElementSelection: \\"rgb(236, 187, 193)\\";
  colorBackgroundError: \\"rgb(206, 80, 95)\\";
  colorBackgroundErrorDark: \\"rgb(174, 33, 49)\\";
  colorBackgroundErrorLight: \\"rgb(194, 37, 55)\\";
  colorBackgroundRowActive: \\"rgb(230, 230, 230)\\";
  colorBackgroundRowFocus: \\"rgb(214, 214, 214)\\";
  colorBackgroundRowHover: \\"rgb(214, 214, 214)\\";
  colorBackgroundRowSelected: \\"rgb(230, 230, 230)\\";
  colorBackgroundRowStriped: \\"rgb(235, 235, 235)\\";
  colorBackgroundSuccess: \\"rgb(100, 209, 138)\\";
  colorBackgroundSuccessDark: \\"rgb(80, 167, 110)\\";
  colorBackgroundSuccessLight: \\"rgb(162, 227, 184)\\";
  colorBackgroundWarning: \\"rgb(251, 208, 144)\\";
  colorBackgroundWarningDark: \\"rgb(199, 141, 56)\\";
  colorBackgroundWarningLight: \\"rgb(249, 177, 71)\\";
};
export declare const colors: {
  colorBrandHighlight: \\"rgb(242, 47, 70)\\";
  colorBrandHighlightDark: \\"rgb(235, 15, 41)\\";
  colorBrandPrimary: \\"rgb(13, 18, 43)\\";
  colorGray1: \\"rgb(255, 255, 255)\\";
  colorGray10: \\"rgb(34, 34, 34)\\";
  colorGray2: \\"rgb(246, 246, 246)\\";
  colorGray3: \\"rgb(235, 235, 235)\\";
  colorGray4: \\"rgb(230, 230, 230)\\";
  colorGray5: \\"rgb(214, 214, 214)\\";
  colorGray6: \\"rgb(204, 204, 204)\\";
  colorGray7: \\"rgb(153, 153, 153)\\";
  colorGray8: \\"rgb(102, 102, 102)\\";
  colorGray9: \\"rgb(68, 68, 68)\\";
};
export declare const fontSizes: {
  fontSize0: \\"0.6875rem\\";
  fontSize1: \\"0.75rem\\";
  fontSize2: \\"0.8125rem\\";
  fontSize3: \\"0.875rem\\";
  fontSize4: \\"0.9375rem\\";
  fontSize5: \\"1rem\\";
  fontSize6: \\"1.125rem\\";
  fontSize7: \\"1.25rem\\";
  fontSize8: \\"1.875rem\\";
  fontSize9: \\"2.25rem\\";
};
export declare const spacings: {
  space0: \\"0.25rem\\";
  space1: \\"0.5rem\\";
  space10: \\"2.75rem\\";
  space11: \\"3rem\\";
  space12: \\"3.25rem\\";
  space13: \\"3.5rem\\";
  space14: \\"3.75rem\\";
  space15: \\"4rem\\";
  space2: \\"0.75rem\\";
  space3: \\"1rem\\";
  space4: \\"1.25rem\\";
  space5: \\"1.5rem\\";
  space6: \\"1.75rem\\";
  space7: \\"2rem\\";
  space8: \\"2.25rem\\";
  space9: \\"2.5rem\\";
};
export declare const textColors: {
  colorTextBrandHighlight: \\"rgb(242, 47, 70)\\";
  colorTextBrandHighlightDark: \\"rgb(235, 15, 41)\\";
  colorTextBrandPrimary: \\"rgb(255, 255, 255)\\";
  colorTextDefault: \\"rgb(68, 68, 68)\\";
  colorTextError: \\"rgb(194, 37, 55)\\";
  colorTextIconDefault: \\"rgb(68, 68, 68)\\";
  colorTextInverse: \\"rgb(255, 255, 255)\\";
  colorTextInverseWeak: \\"rgb(235, 235, 235)\\";
  colorTextLabel: \\"rgb(68, 68, 68)\\";
  colorTextLink: \\"rgb(0, 112, 204)\\";
  colorTextLinkActive: \\"rgb(0, 89, 163)\\";
  colorTextLinkDestructive: \\"rgb(194, 37, 55)\\";
  colorTextLinkDestructiveActive: \\"rgb(155, 29, 44)\\";
  colorTextLinkDestructiveDisabled: \\"rgb(236, 187, 193)\\";
  colorTextLinkDestructiveFocus: \\"rgb(174, 33, 49)\\";
  colorTextLinkDestructiveHover: \\"rgb(174, 33, 49)\\";
  colorTextLinkDisabled: \\"rgb(102, 169, 224)\\";
  colorTextLinkFocus: \\"rgb(0, 97, 192)\\";
  colorTextLinkHover: \\"rgb(0, 97, 192)\\";
  colorTextPlaceholder: \\"rgb(102, 102, 102)\\";
  colorTextSuccess: \\"rgb(100, 209, 138)\\";
  colorTextWarning: \\"rgb(249, 177, 71)\\";
  colorTextWeak: \\"rgb(230, 230, 230)\\";
};
export declare const zIndices: {
  zIndex0: 0;
  zIndex10: 10;
  zIndex20: 20;
  zIndex30: 30;
  zIndex40: 40;
  zIndex50: 50;
  zIndex60: 60;
  zIndex70: 70;
  zIndex80: 80;
  zIndex90: 90;
};"
`;
